File: CS50x/pset6/caesar.py
import sys
import cs50

# Allow two arguments only, program will exit otherwise
if len(sys.argv) != 2:
    # change error message
    print("Please provide one argument only")
    exit(1)

# Use the the second argument as key
k = int(sys.argv[1])

# Print instructions on screen, ask for input and calculate the length of it
print("plaintext: ", end="")
ptxt = cs50.get_string()
n = len(ptxt)
print("ciphertext: ", end="")

# Iterate over each char
for c in ptxt:

# If the current char is in lowercase, rotate by k positions. Alpha is used to set a alphabetical order starting from zero
    if str.islower(c):
        alpha = ord(c) - 97
        print(chr((alpha + k) % 26 + 97), end="")

# If the current char is in uppercase, find its numeric position on alphabet, rotate by k positions and print it. Modulo is used so Z becomes A.
    if str.isupper(c):
        print(chr((ord(c) - 65 + k ) % 26 + 65), end="")

# If the i'th char is not a letter print it whithout unchanged.
    if not str.isalpha(c):
        print(c, end="")

# Print new line (layout)
print()

File: CS50x/pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main (int argc, string argv[])
{
    // If no argument is provided, return exit code 1
    if (argc < 2 || argc > 2)
    {
        printf("Please use one argument\n"); 
        return 1;
    }
    
// Ask for plain text    
printf("plaintext: ");
string ptxt = get_string();

// Convert string to int
int k = atoi(argv[1]);

int n = strlen(ptxt);

printf("ciphertext: ");    
    // Iterates over each character
    for (int i = 0; i <= n; i++)
    {
        // Ckeck if the i'th character is in lowercase
        if (islower (ptxt[i]))
        {
            // Define alphabetical order (alpha == position in alphabet minus ASCII code for the first letter - in lowercase or uppercase)
            int alpha = (int)ptxt[i] - 97;
            // Convert char to ASCII again and print the char rotation by k positions using the algorithm C = (p + k) % 26 
            printf("%c", (alpha + k) % 26 + 97);
        }
        // Check if the i'th char is in uppercase
        if (isupper (ptxt[i]))
        {
            // Same as above but in one line. I'th char - 65 (Get alphabetical position) rotated by k mod 26 + 65 (Convert back to ASCII)
            printf("%c", ( (int)ptxt[i] - 65 + k ) % 26 + 65);
        }
        // Check if the char is a space or punctuation and print without any modification
        if( isspace(ptxt[i]) || ispunct(ptxt[i]) )
        {
            printf("%c", ptxt[i]);
        }
        
    }
    printf("\n");
}